CIP-651 #time 5h > Add: created Api for user management module > Also…

codal-rutvik opened the pull request:
… created a middleware for protected route

codal-rutvik committed:
CIP-651 #time 5h > Add: created Api for user management module > Also created a middleware for protected route

codal-rutvik committed:
CIP-651 #time 4h > Add: User Profile api for get and update profile > Refactor: Phone number field validation

codal-rutvik committed:
CIP-651 time #20m > Refactor: Remove the password field from get profile user api response > Updated the function as per JS ES6 syntax

rahulr-codal commented on app/routes/auth.js:
```diff
@@ -0,0 +1,11 @@
+
+router.post("/signup", userController.signup);
+router.post("/login", userController.login);
+router.get("/user/profile", authMiddleware, userController.getUserProfile);
+router.put("/user/profile", authMiddleware, userController.updateUserProfile);
```
this will be like -> `auth/user/profile`.
Think of some other route as auth works good with signup and login but not with getting users related stuff.

rahulr-codal commented on app/models/user.model.js:
```diff
@@ -0,0 +1,28 @@
+
+const User = mongoose.model('User', userSchema);
+
+module.exports = User;
```
rename this from user.model to user.js only as file naming should not contain `.`

rahulr-codal commented on app/controllers/userController.js:
```diff
@@ -0,0 +1,190 @@
+  });
+
+  try {
+    // Validate user input against the schema
+    const { error } = schema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error?.details[0]?.message });
+    }
+
```
create one common module for validation of joi, so that you can remove redundancy of getting error message  and validating based on each type. The new function should take options and give you either validated data or error. This will be used/share by all you different APIs

rahulr-codal commented on app/controllers/userController.js:
```diff
@@ -0,0 +1,190 @@
+
+    res.status(200).json({ message: "User registered successfully" });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "An error occurred during signup" });
+  }
```
For error use one error handler on index.js file. 
from here error should be send using next(err) and your error middleware will take care of sending status and message.

Also, make sure to pass the statusfor different error which are sending here.

rahulr-codal replied:
do this for all the API below

codal-rutvik committed:
update: added common error middleware

CIP-651 #time 2h sperate an auth and user controller > update joi validation and added common module

rahulr-codal commented on app/routes/user.js:
```diff
@@ -0,0 +1,9 @@
+const userController = require("../controllers/userController");
+const authMiddleware = require("../middleware/authMiddleware");
+
+router.get("/profile", authMiddleware, userController.getUserProfile);
```
we discussed to remove  `/profile` as it doesn't make sense, right?

codal-rutvik replied:
So you mean need to just keep **/user** right?

rahulr-codal replied:
yess

codal-rutvik replied:
Updated.

rahulr-codal commented on app/routes/user.js:
```diff
@@ -0,0 +1,9 @@
+const authMiddleware = require("../middleware/authMiddleware");
+
+router.get("/profile", authMiddleware, userController.getUserProfile);
+router.put("/profile", authMiddleware, userController.updateUserProfile);
```
we discussed to remove  `/profile` as it doesn't make sense, right?

codal-rutvik committed:
update: api routes for the user module

CIP-651 #time 15m > Updated the api route for the user module api for better api route convention

codal-rutvik committed:
update: updated the user model file name

CIP-651 #time > Updated the user model file name and remove the . between the file name

codal-rutvik committed:
Remove the user.model.js file
